### PR TITLE
Add a regression test for memrchr errors with clang

### PR DESCRIPTION
**Pull request by lazka**
See https://github.com/msys2/MINGW-packages/pull/13125

### CODE REVIEW

File: toolchain/meson/tests/meson.build [hunk 1]
```diff
@@ -16,3 +16,5 @@ windres = executable('windres', 'windres.c', windres_res)
 test('windres', windres)
 
 test('call_once', executable('call_once', 'call_once.cpp'))
+
+test('undef_memrchr', executable('undef_memrchr', 'undef_memrchr.c', c_args: ['-O3']))
```
Comment (1480c1) [inline toolchain/meson/tests/meson.build:20]: technically, `-O1` is good enough

**Comment by mati865** [comment]
It's very unlikely to regress the same way again but other regression on other symbols is more likely 😞

File: toolchain/meson/tests/meson.build (hunk 1 above)
Comment (lazka) [reply toolchain/meson/tests/meson.build:20]: but can't hurt, no?

**Comment by lazka** [comment]
> It's very unlikely to regress the same way again but other regression on other symbols is more likely 😞

Yeah... just trying to make a habit of adding tests..

**Pull request merged**